### DISCUSSION TITLE
Referring to a bibliography in another file without showing it

### DISCUSSION THREAD

**Discussion by awa5114** [jgm/pandoc#10147] [Q&A]
I am trying to build a simple static website with Pandoc-generated HTML files. The site has a main page generated from `index.md`. This main page contains links to other pages corresponding to various topics. Here’s a sample of what the markdown looks like:

```
---
title: "Main page"
---

- [Topic1](topic1.html)
- [Topic2](topic2.html)
- ....
- [References](references.html)
```

Within the various topic pages, I would like to include citations such as `[@my-citation]`, which would refer to any one of a number of citations stored in a `references.json` file in the root directory.

Here’s the catch though, I don’t want to generate a (visible) bibliography in each topic page. I want to generate the bibliography once in `references.html`, but have the citations in all the topic pages *refer* and *link*  to this same bibliography using the `ieee.csl` style.

I was able to successfully generate a lone bibliography as follows:

`pandoc .\references.json --metadata title:References --citeproc -o references.html --csl ieee.csl -f csljson -t html`

However I’m not sure how to prevent the bibliography from appearing on each topic page, because using `--bibliography references.json` and the `--citeproc` flag will result in a bibliography on the page currently being compiled.

Is there a flag or some workaround to prevent the bibliography from appearing in each topic page?

**Comment by awa5114** [comment]
I just realized I can use `suppress-bibliography: true` in the metadata for the topic pages, but then `link-citations: true` appears to have no effect

**Reply by jgm** [reply]
You'll only get links to a bibliography in the same document, sorry.

**Reply by jgm** [reply]
As long as the ids on the bibliography page are predictable from the citation ids, you should be able to create a Lua filter that adds the links you need to citations.

**Reply by awa5114** [reply]
Thanks @jgm. The ids on the bibliography page appear to be identical to the citation ids, so I'll work on a filter that converts them to links and share it on here if I'm successful.

**Comment by wlupton** [comment]
This might be more complex than you want, but an alternative is to use a single file (assuming it's not so large that this is impracticable) and use JavaScript to hide/show sections in order to give the _appearance_ of a multi-file site. See https://usp.technology/specification for an example (it's all done with lua filters).

**Comment by wlupton** [comment]
(Of course you can still have your markdown in multiple files (and we do) but they all come together for the pandoc run.)
